bubuku/cli.py

```diff
@@ -204,5 +208,65 @@ def show_stats():
+    with load_exhibitor_proxy(env_provider.get_address_provider(), config.zk_prefix) as ex:
+        def get_broker_ids_sorted_by_load(broker_ids) -> []:
+            load_by_broker = {
+                id_: float(requests.get(
+                    ':8778/jolokia/read/java.lang:type=OperatingSystem'.format(ex.get_broker_address(id_))).json()[
+                               'value']['SystemCpuLoad'])
+                for id_ in broker_ids
+                }
+            print('Brokers load: {}'.format(load_by_broker))
+            return sorted(broker_ids, key=lambda id_: load_by_broker[id_])
+
+        def get_topics_load_sorted(broker_id, tps) -> dict:
+            ip = ex.get_broker_address(broker_id)
+            load_dict = {}
+            for topic, partition in tps:
+                if topic in load_dict:
+                    continue
+                load_dict[topic] = requests.get(
+                    '{}:8778/jolokia/read/kafka.server:name=MessagesInPerSec,type=BrokerTopicMetrics,topic={}'
+                        .format(ip, topic)).json()['value']['OneMinuteRate']
+            return sorted(tps, key=lambda tp: load_dict.get(tp[0]))
+
+        def select_swap_candidate(loaded_tps, not_loaded_tps) -> tuple:  # topic partition
+            for i in range(0, len(loaded_tps) // 10):
+                candidate = loaded_tps[len(loaded_tps) - 1]
+                if candidate in not_loaded_tps:
+                    return candidate
+
+        tp_to_brokers = {}
+        leadership_broker_to_tp_list = {}
+        for topic, partition, brokers in ex.load_partition_assignment():
+            tp = (topic, partition)
+            tp_to_brokers[tp] = brokers
+
+            leader = brokers[0]
+            if leader not in leadership_broker_to_tp_list:
+                leadership_broker_to_tp_list[leader] = []
+            leadership_broker_to_tp_list[leader].append(tp)
+        load_by_leader = {}
+        for leader, tps in leadership_broker_to_tp_list.items():
+            load_by_leader[leader] = get_topics_load_sorted(leader, tps)
+        load_ordered = get_broker_ids_sorted_by_load()
+        loaded = load_ordered[len(load_ordered) - 1]
+        for not_loaded in load_ordered[:len(load_ordered) // 2]:
+            candidate = select_swap_candidate(load_by_leader[loaded], load_by_leader[not_loaded])
+            if candidate:
+                broker_ids = tp_to_brokers[candidate]
+                print('old allocation for {}: {}'.format(candidate, broker_ids))
+                length = len(broker_ids)
+                del broker_ids[not_loaded]
+                broker_ids = [not_loaded] + broker_ids
+                if length != len(broker_ids):
+                    raise Exception('broker ids length changed')
+                print('New allocation for candidate {}: {}'.format(candidate, broker_ids))
+                # ex.reallocate_partition(candidate[0], candidate[1], broker_ids)
+                return
```

Comment:
I'm going to remove this entirely, if we want to merge it from 'migrate-data', it should be in a different PR (and with tests :) )